pandas/core/generic.py

```diff
@@ -10932,6 +10932,12 @@ def _add_numeric_operations(cls):
         """
         axis_descr, name1, name2 = _doc_params(cls)
 
+        @deprecate_nonkeyword_arguments(
+            version=None,
+            allowed_args=["self"],
+            stacklevel=find_stack_level() - 1,
```

Comment:
removing the stacklevel from args (keeping default value of 2) also works here 🤔

Reply:
its not correct to evaluate this at compile time. what should happen is that the deprecater can *call* `find_stack_level`

Reply:
@jreback I don't understand what you mean, how should this be?

Just setting `stacklevel=find_stack_level()` doesn't work because `any` is inside `_add_numeric_operations` and is set via `setattr(cls, "any", any)`

https://github.com/pandas-dev/pandas/blob/073b3535d7a5171102e5915c38b57c21d13795ae/pandas/core/generic.py#L10928-L10948

Reply:
i mean this needs to be handled *in the decorator itself* not here.

Reply:
It's done in the decorator in other places though

https://github.com/pandas-dev/pandas/blob/e1b95fa128203e6e21d3a1a6469c94e49f503e0c/pandas/io/parsers/readers.py#L831-L833

I've made an issue to handle this in the decorator itself https://github.com/pandas-dev/pandas/issues/46687 , is it OK to keep that separate from this PR?

Reply:
ok i c, yeah we out to fix this but ok for here